test/Microsoft.ML.Functional.Tests/Datasets/AllTypes.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Data.DataView;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Xunit;
+
+namespace Microsoft.ML.Functional.Tests.Datasets
+{
+    internal sealed class AllTypes
+    {
+        [LoadColumn(0)]
+        public bool Label { get; set; }
+
+        [LoadColumn(1, 5), VectorType(5)]
+        public float[] Features { get; set; }
+
+        [LoadColumn(6)]
+        public sbyte I1 { get; set; }
+
+        [LoadColumn(7)]
+        public byte U1 { get; set; }
+
+        [LoadColumn(8)]
+        public short I2 { get; set; }
+
+        [LoadColumn(9)]
+        public ushort U2 { get; set; }
+
+        [LoadColumn(10)]
+        public int I4 { get; set; }
+
+        [LoadColumn(11)]
+        public uint U4 { get; set; }
+
+        [LoadColumn(12)]
+        public long I8 { get; set; }
+
+        [LoadColumn(13)]
+        public ulong U8 { get; set; }
+
+        [LoadColumn(14)]
+        public float R4 { get; set; }
+
+        [LoadColumn(15)]
+        public double R8 { get; set; }
+
+        [LoadColumn(16)]
+        public ReadOnlyMemory<char> Tx { get; set; }
+
+        [LoadColumn(17)]
+        public TimeSpan Ts { get; set; }
+
+        [LoadColumn(18)]
+        public DateTime Dt { get; set; }
+
+        [LoadColumn(19)]
+        public DateTimeOffset Dz { get; set; }
+
+        [LoadColumn(20)]
+        public RowId Ug { get; set; }
+
+        /// <summary>
+        /// Get the text loader for the AllTypes dataset.
+        /// </summary>
+        /// <param name="mlContext">The ML Context.</param>
+        /// <param name="separator">The Separator to read with.</param>
+        /// <returns></returns>
+        public static TextLoader GetTextLoader(MLContext mlContext, char separator)
+        {
+            return mlContext.Data.CreateTextLoader(
+                    new[] {
+                        new TextLoader.Column("Label", DataKind.Bool, 0),
+                        new TextLoader.Column("Features", DataKind.R4, 1, 5),
```

Comment:
>5 [](start = 74, length = 1)

factor out as param.  Do same with the GetRandomINstance()

Reply:
I can't add this as a factor unless there is a way to alter the class attributes dynamically. For our tests, we need LoadColumn attributes to point to the corresponding serialization column.

---
In reply to: [256542590](https://github.com/dotnet/machinelearning/pull/2518#discussion_r256542590) [](ancestors = 256542590)